### DISCUSSION TITLE
Help: How do we add dependencies manually?

### DISCUSSION THREAD

**Discussion by abhijit-inge** [google/or-tools#3507] [Build and configuration issues]
I tried building the package locally using cmake and it builds without any errors. However the environment which we are planning to ultimately use does not have an outbound network connection. The problem I see is that https://github.com/google/or-tools/blob/v9.4/cmake/dependencies/CMakeLists.txt performs a Git Clone to download the dependencies and add them. Since there is no outbound network access this step fails and I'm unable to build the dependency. To circumvent this we are planning to manually download the dependencies and add them to https://github.com/google/or-tools/blob/v9.4/cmake/dependencies/ folder. 
I'm pretty new to CMake and I'm not sure what changes have to be made to accommodate this. Could you help us out?

For example, what would the following snippet need to be changed to if I added Zlib v1.2.11 repository to https://github.com/google/or-tools/blob/v9.4/cmake/dependencies/? Any help would be appreciated. Thanks!

```
# ##############################################################################
# ZLIB
# ##############################################################################
if(BUILD_ZLIB)
  message(CHECK_START "Fetching ZLIB")
  list(APPEND CMAKE_MESSAGE_INDENT "  ")
  FetchContent_Declare(
    zlib
    GIT_REPOSITORY "https://github.com/madler/ZLIB.git"
    GIT_TAG "v1.2.11"
    PATCH_COMMAND git apply --ignore-whitespace "${CMAKE_CURRENT_LIST_DIR}/../../patches/ZLIB.patch")
  FetchContent_MakeAvailable(zlib)
  list(POP_BACK CMAKE_MESSAGE_INDENT)
  message(CHECK_PASS "fetched")
endif()
```

**Comment by lperron** [comment]
why don't you use the packaged libraries, or build your own on a machine with network access?

**Comment by abhijit-inge** [comment]
Unfortunately machine with network access isn't an option for me :(. How do i use the packaged libraries? 
I'm guessing all the files that are already created under `builds/lib64 & build/_deps` have to be copied over and set the BUILD_DEPS:BOOL flag to OFF? Any other files need to be copied over?

**Comment by lperron** [comment]
If you can install the source, you can install the packaged libraries.

Or am I missing something.

Le mer. 19 oct. 2022, 20:23, ABHIJIT INGE ***@***.***> a
écrit :
